fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckVendorsService.kt

```diff
@@ -54,6 +59,8 @@ class DefaultReviewQualityCheckVendorsService(
         }
     }
 
+    override fun getProductVendor(): ProductVendor = productVendors().first()
```

Comment:
Do we need this? I think if we're just doing first, we could do that in the middleware and keep the `RQCVendorsService` API lean, wdyt?

Reply:
I guess my intent was to abstract the implementation away from the Middleware, since it likely shouldn't know that `productVendors()` is an ordered list with the product's vendor first. 

`productVendors().first()` wasn't the first iteration of this. I initially looked at doing something similar at the Gecko mapping process, but then discovered we need the vendor at the entire `OptedIn` section of the feature, so my options became:
1. Refactor `productVendors` and have some shared logic between two function calls
2. Write `getProductVendor` completely from scratch but with the assumption that it has a good-formed URL
3. Create `getProductVendor` and tack on `first` because of the assumptions we're making about correctness. 
4. Do `productVendors().first()` inside of Middleware and call it a day.

I opted for number 3 in order to keep the implementation simple and the Middleware blinded to the internals of either of the function.

Reply:
We could also "guarantee" that the first is always the relevant one with tests and kdocs though

Reply:
4 seems good to me as we use the existing API and that's tested in Store without faking another function but I also see your reasons for 3 where we hide the implementation details 👍